payments-core/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -182,9 +182,12 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
 
     fun getSupportedPaymentMethods(): List<SupportedPaymentMethod> {
         stripeIntent.value?.let { stripeIntent ->
-            return stripeIntent.paymentMethodTypes.mapNotNull {
+            return stripeIntent.paymentMethodTypes.filter {
```

Comment:
nit: 
`config?.supportsDelayedSettlement || !PaymentMethod.Type.fromCode(it)?.hasDelayedSettlement()`

Reply:
This won't work because the checks need to account for null values.